packages/lib-core/src/extensions/actions.ts

```diff
@@ -0,0 +1,119 @@
+import type { ReactNode } from 'react';
+import type { ExtensionHook, ExtensionK8sKindVersionModel } from '../api/common-types';
+import type { CodeRef, Extension } from '../types/extension';
+import type { AccessReviewResourceAttributes } from './core-types';
+
+/** ActionProvider contributes a hook that returns list of actions for specific context */
+export type ActionProvider = Extension<
+  'core.action/provider',
+  {
+    /** The context ID helps to narrow the scope of contributed actions to a particular area of the application. Ex - topology, helm */
+    contextId: string | 'resource';
+    /** A react hook which returns actions for the given scope.
+     * If contextId = `resource` then the scope will always be a K8s resource object
+     * */
+    provider: CodeRef<ExtensionHook<Action[]>>;
```

Comment:
I'll look into all `ExtensionHook<T>` type references as part of a second round of code review.

I'm not sure why we need to reference the signature of the extension consuming hook itself, I'd assume this is primarily due to type/signature commonality in which case we should introduce a type with a more suitable name.

@andrewballantyne FYI this needs looking into.

Reply:
> as part of a second round of code review

Can be done after merging this PR, I don't want to hold it on this.

@andrewballantyne My concern was why do we even need to know the signature of `useResolvedExtension` hook (function) when dealing with extension-related types :smiley: is it because of structural similarity, or other reason? As I wrote, we can improve this later on.

Reply:
To elaborate, extension-related types (i.e. ones referenced by specific `Extension` declarations) are used to _describe_ specific extension points in the application; basically using data and/or callback functions.

I'm missing the logical link to `useResolvedExtension` which is a function used to consume/interpret extensions.

Reply:
OK just realized it's because we want to pass React hook via `provider` property.